agent/internal/util/json.go

```diff
@@ -7,7 +7,7 @@ import (
 func RemoveJSONComment(strIn []byte) []byte {
 	str := string(strIn)
 	// edgecase that isn't handled: double quote in the comment
-	re1 := regexp.MustCompile(`//\s*?[^"\\S]*?\s*?(\n|\r|\r\n|\n\r)`)
+	re1 := regexp.MustCompile(`([^\S\r\n]*)//\s*?[^"\\S]*?\s*?(\n|\r|\r\n|\n\r)`)
```

Comment:
Regexes are generally difficult to read, I don't see the added value, could you please explain?
`\S = ^\s` so this at the beginning is a double negation `^\S = ^^\s = \s`?

Reply:
Of course, this group means, as you mention, double negate whitespace + negate newline chars. I am sure it could be done differently, but this in the end means "strip all whitespace characters except newlines".

I did try `\s` and what was happening is that new lines were stripped before the comment (not just at the end). This would result in lines being put all together in one.